app/models/user.rb

```diff
@@ -0,0 +1,58 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :confirmable, :trackable,
+         :omniauthable, omniauth_providers: %i[facebook]
+
+  attr_accessor :school_year
+
+  before_save :calculate_graduation_year
+
+  validates :first_name, :last_name, :username, presence: true
+  validates_uniqueness_of :username
+
+  enum user_type: { student: 0, teacher: 1, admin: 2 }
```

Comment:
make admin as 0 so it's easy to remember, then 1 is student who are the most common and then teacher is 2.